lntest/itest/lnd_routing_test.go

```diff
@@ -1175,6 +1175,15 @@ func testInvoiceRoutingHints(net *lntest.NetworkHarness, t *harnessTest) {
 	}
 	checkInvoiceHints(ctxb, t.t, net, aliceBobChanID, invoice)
 
+	// Create another invoice for Alice with no value and ensure it still
+	// populates routing hints.
+	invoice = &lnrpc.Invoice{
+		Memo:    "routing hints with no amount",
+		Value:   0,
+		Private: true,
+	}
+	checkInvoiceHints(ctxb, t.t, net, aliceBobChanID, invoice)
```

Comment:
This new structure is so much better for review, thanks!

As a final comment, I'd squash the final three commits. So in one commit, add the fix and update the tests. During review, it is nice to cross check both without going back and forth commits.

Reply:
Sounds good, I've just pushed that squashing :+1: